touchforms/formplayer/static/formplayer/script/fullform-ui.js

```diff
@@ -1,634 +1,459 @@
-    return ix;
-  } else {
-    var rel_ix = '-';
-    for (var i = deepest_repeat + 1; i < steps.length; i++) {
-      rel_ix += steps[i] + (i < steps.length - 1 ? ',' : '');
+    if (deepest_repeat == -1) {
+        return ix;
+    } else {
+        var rel_ix = '-';
+        for (var i = deepest_repeat + 1; i < steps.length; i++) {
+            rel_ix += steps[i] + (i < steps.length - 1 ? ',' : '');
+        }
+        return rel_ix;
     }
-    return rel_ix;
-  }
 }
 
 function getIx(o) {
-  var ix = o.rel_ix;
-  while (ix[0] == '-') {
-    o = o.parent;
-    if (!o) {
-      break;
-    }
-    if (o.rel_ix.split(',').slice(-1)[0].indexOf(':') != -1) {
-      ix = o.rel_ix + ',' + ix.substring(1);
+    var ix = ko.utils.unwrapObservable(o.rel_ix);
+    while (ix[0] == '-') {
+        o = o.parent;
+        if (!o) {
+            break;
+        }
+        if (ix.split(',').slice(-1)[0].indexOf(':') != -1) {
+            ix += ',' + ix.substring(1);
+        }
     }
-  }
-  return ix;
+    return ix;
 }
 
 function getForIx(o, ix) {
-  if (o.type == 'question') {
-    return (getIx(o) == ix ? o : null);
-  } else {
-    for (var i = 0; i < o.children.length; i++) {
-      var result = getForIx(o.children[i], ix);
-      if (result) {
-        return result;
-      }
-    }
-  }
-}
-
-function ixInfo(o) {
-  var full_ix = getIx(o);
-  return o.rel_ix + (o.is_repetition ? '(' + o.uuid + ')' : '') + (o.rel_ix != full_ix ? ' :: ' + full_ix : '');
-}
-
-function empty_check(o, anim_speed) {
-  if (o.type === 'repeat-juncture') {
-    if (anim_speed) {
-      o.$empty[o.children.length ? 'slideUp' : 'slideDown'](anim_speed);
+    if (o.type == 'question') {
+        return (getIx(o) == ix ? o : null);
     } else {
-      o.$empty[o.children.length ? 'hide' : 'show']();
-    }
-  } else if (o.type === 'sub-group') {
-    if (anim_speed) {
-      o.$container[o.children.length ? 'slideDown' : 'slideUp'](anim_speed);
-    } else {
-      o.$container[o.children.length ? 'show' : 'hide']();
+        for (var i = 0; i < o.children.length; i++) {
+            var result = getForIx(o.children[i], ix);
+            if (result) {
+                return result;
+            }
+        }
     }
-  }
 }
 
-function loadFromJSON(o, json) {
-  $.each(json, function(key, val) {
-      if (key == 'children') {
-        return;
-      } else if (key == 'ix') {
-        key = 'rel_ix';
-        val = relativeIndex(val);
-      } else if (key == 'answer') {
-        key = 'last_answer';
-      } else if (key == 'style') {
-        if ('domain_meta' in json) {
-          key = 'domain_meta';
-          val = parse_meta(json.datatype, val);
-        } else {
-          key = 'style';
-        }
-      }
-
-      o[key] = val;
-    });
+function ixInfo(o) {
+    var full_ix = getIx(o);
+    return o.rel_ix + (o.isRepetition ? '(' + o.uuid + ')' : '') + (o.rel_ix != full_ix ? ' :: ' + full_ix : '');
 }
 
 function parse_meta(type, style) {
```

Comment:
my guess is most of this is relics from other UI modalities (touchscreen, typeahead)